### DISCUSSION TITLE
could it add a convert function()  which make input is a Mat format

### DISCUSSION THREAD

**Discussion by hcgprague** [tshino/softcam#35] [General]
could it add a convert function()  which make input is a Mat format ?
a convert function()  which make input is a Mat format  as :

void  Mat2DWimage( Mat matImage,    vector<unsigned char>  &directShowImage  )
{
int width = matImage.cols;
int height = matImage.rows;
int channels = matImage.channels();
std::vector<unsigned char> directShowImage(width * height * 3);
if (matImage.isContinuous()) {
    memcpy(directShowImage.data(), matImage.data, directShowImage.size());
} else {
    // If the image is not stored continuously in memory, copy row by row
    for (int row = 0; row < height; ++row) {
        memcpy(&directShowImage[row * width * 3], matImage.ptr(row), width * 3);
    }
}
}

**Comment by tshino** [comment]
@hcgprague Hi!, thank you for proposing a useful code.

I guess the `Mat` class comes from OpenCV, so your code would be helpful if one is using both Softcam and OpenCV.
Adding such code to Softcam library will introduce an unnecessary library dependency for users who don't use OpenCV.
So I won't add this code to Softcam library itself.

But I believe that as long as your code is valid it can help other people who want to do the same.
So, I enabled the Discussions tab of this GitHub repo, where I think your post could keep open to be found for someone finding information.

I will look at applying **Convert to discussion** feature to this issue.

**Comment by wukongxuetang** [comment]
This code is ok！

uint8_t* Mat_to_array(const cv::Mat input)
{
    int height = input.rows;
    int width = input.cols;
    uint8_t* pRgb = new uint8_t[height * width * 3];
    for (int i = 0;i < height;i++)
    {
        for (int j = 0;j < width;j++)
        {
            for (int k = 0;k < 3;k++)
            {
                pRgb[i * width * 3 + j * 3 + k] = input.at<cv::Vec3b>(i, j)[k];
            }
        }
    }
    return pRgb;
}

use it just like below:
inFrame = getImage.clone();
if(scWaitForConnection(scam),0.001f) scSendFrame(scam,Mat_to_array(inFrame));

**Reply by wukongxuetang** [reply]
this code can be used in sender,not for softcam dll.
